var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -197,6 +197,8 @@ class Llvm(CMakePackage, CudaPackage):
     # Introduced in version 11 as a part of LLVM and not a separate package.
     conflicts("+flang", when="@:10")
 
+    conflicts('~mlir', when='+flang', msg='flang implies MLIR')
```

Comment:
Mind tweaking the wording? Otherwise I'm all good with this.
```suggestion
    conflicts('~mlir', when='+flang', msg='flang requires MLIR')
```

Reply:
Thanks!